operator/src/main/java/oracle/kubernetes/operator/helpers/ResponseStep.java

```diff
@@ -135,16 +135,13 @@ protected final NextAction doContinueListOrNext(CallResponse<T> callResponse, Pa
    * @return Next action for retry or null, if no retry is warranted
    */
   private NextAction doPotentialRetry(Step conflictStep, Packet packet, CallResponse<T> callResponse) {
-    RetryStrategy retryStrategy = packet.getSpi(RetryStrategy.class);
-    if (retryStrategy != null) {
-      return retryStrategy.doPotentialRetry(conflictStep, packet, callResponse.getStatusCode());
-    }
-
-    LOGGER.fine(MessageKeys.ASYNC_NO_RETRY,
-        callResponse.getExceptionString(),
-        callResponse.getStatusCode(),
-        callResponse.getHeadersString());
-    return null;
+    return Optional.ofNullable(packet.getSpi(RetryStrategy.class))
+        .map(rs -> rs.doPotentialRetry(conflictStep, packet, callResponse.getStatusCode()))
+        .orElseGet(() -> {
```

Comment:
Probably I'm missing something but I'm not able to see how this is different from previous the code i.e. how it logs the failures when the retry strategy is not null?

Reply:
If I understand this correctly, the only difference is that when doPotentialRetry returns null (which means no need to retry), the old code does not log and the new code does.

Reply:
ok, got it now. Thanks Dongbo.